browser/src/index.ts

```diff
@@ -0,0 +1,103 @@
+
+function unpackU64toU8(n: BigInt): Uint8Array {
+    //@ts-ignore
+    return new Uint8Array(new BigInt64Array([n]).buffer)
+}
+
+function packU8toU64(bytes: Uint8Array): BigInt {
+    return (new BigInt64Array(bytes.buffer))[0]
+}
+
+function stringToBytes(s: string, offset: number, len: number): Uint8Array {
+    return new Uint8Array(s.slice(offset, len).split("").map(c => c.charCodeAt(0)))
+}
+
+function makeEnv(plugin: ExtismPlugin): any {
+    return {
+        extism_alloc(n: BigInt): BigInt {
+            return BigInt(50) // just pick a safe? number for now
+        },
+        extism_free (n: BigInt) {debugger},
+        extism_load_u8(n: BigInt): number { debugger; return 0 },
+        extism_load_u32(n: BigInt): number {debugger; return 0 },
+        extism_load_u64(n: BigInt): BigInt { debugger;return BigInt(0)},
+        extism_store_u8(offset: BigInt, n: number) {
+            //@ts-ignore
+            plugin.memory[offset] = Number(n)
+        },
+        extism_store_u32(n: BigInt, i: number) {debugger; },
+        extism_store_u64(offset: BigInt, n: BigInt) {
+            let parts = unpackU64toU8(n)
+            parts.forEach((p, i) => {
+                plugin.memory[i + Number(offset)] = p
+            })
+        },
+        extism_input_length(): BigInt {
+            //@ts-ignore
+            return BigInt(plugin.input.length)
+        },
+        extism_input_load_u8(i: number): number {
+            //@ts-ignore
+            return plugin.input[i].charCodeAt(0)
+        },
+        extism_input_load_u64(idx: BigInt) {
+            const i = Number(idx)
+            //@ts-ignore
+            return packU8toU64(stringToBytes(plugin.input, i, i+8))
+        },
+        extism_output_set(offset: BigInt, len: number): number {
+            //@ts-ignore
+            offset = Number(offset)
+            len = Number(len)
+            //@ts-ignore
+            plugin.output = String.fromCharCode(...plugin.memory.slice(offset, offset+len))
```

Comment:
we probably want to keep the output and input as raw bytes and let the programmer decide if they represent strings or not